libbuteosyncfw/common/Logger.cpp

```diff
@@ -21,215 +22,38 @@
  *
  */
 
+#include <QScopedPointer>
 
 #include "Logger.h"
-#include <QMutexLocker>
-#include <iostream>
-#include <QFile>
-#include <QFileInfo>
-#include <QDir>
 
-#include <syslog.h>
+#define TRACE_FUNCTION_TIME 0
 
 using namespace Buteo;
 
-const int Logger::DEFAULT_INDENT_SIZE = 4;
-
-Logger *Logger::sInstance = nullptr;
-
-Logger *Logger::instance()
-{
-    if (nullptr == sInstance) {
-        createInstance();
-    }
-    return sInstance;
-}
-
-int Logger::defaultLogLevel()
-{
-    QByteArray levelByteArray = qgetenv("MSYNCD_LOGGING_LEVEL");
-    QString levelStr = QString::fromLocal8Bit(levelByteArray.constData());
-    bool ok = false;
-    int level = levelStr.toInt(&ok);
-    if (!ok) {
-        level = 4;  // warning level
-    }
-    return level;
-}
-
-void Logger::createInstance(const QString &aLogFileName, bool aUseStdOut,
-                            int aIndentSize)
-{
-    deleteInstance();
-    sInstance = new Logger(aLogFileName, aUseStdOut, aIndentSize);
-}
-
-void Logger::deleteInstance()
-{
-    delete sInstance;
-    sInstance = nullptr;
-}
-
-Logger::Logger(const QString &aLogFileName, bool aUseStdOut, int aIndentSize)
-    :   iEnabledLevels(NUM_LEVELS),
-        iIndentLevel(0),
-        iIndentSize(aIndentSize),
-        iFileStream(0),
-        iStdOutStream(0),
-        iStdErrStream(0),
-        iEnabled(false),
-        iLogLevel(0)
+LogTimer::LogTimer(const QString &categoryName, const QString &func)
+    : m_categoryName(categoryName.toUtf8())
+    , m_func(func)
+    , m_category(m_categoryName.constData())
 {
-    iLogLevel = defaultLogLevel();
-
-    if (aUseStdOut) {
-        // Create a stream for writing log messages to standard output.
-        iStdOutStream = new QTextStream(stdout);
-    }
-
-    iStdErrStream = new QTextStream(stderr);
-
-    if (!aLogFileName.isEmpty()) {
-        // Create stream for writing log messages to a file.
-        iFile.setFileName(aLogFileName);
-        QFileInfo fileInfo(iFile);
-        QDir dir;
-        dir.mkpath(fileInfo.absolutePath());
-        if (iFile.open(QIODevice::WriteOnly | QIODevice::Append)) {
-            iFileStream = new QTextStream(&iFile);
-        }
-    }
+    qCDebug(m_category) << m_func << ":Entry";
+    m_timer.start();
 }
 
-Logger::~Logger()
+LogTimer::~LogTimer()
 {
-    delete iFileStream;
-    iFileStream = 0;
-
-    delete iStdOutStream;
-    iStdOutStream = 0;
-
-    delete iStdErrStream;
-    iStdErrStream = 0;
+    qCDebug(m_category) << m_func << ":Exit, execution time:" << m_timer.elapsed() << "ms";
```

Comment:
I wonder if ns would be more useful than ms for most functions

Reply:
Would anything in ns level matter here, though?

Reply:
You're right, after thinking about it more, I agree that it's better as ms.  At least we can then easily visually ignore anything which is 0 or 1 ms, and only pay attention to the larger numbers.